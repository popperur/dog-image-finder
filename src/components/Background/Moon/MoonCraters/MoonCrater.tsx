import { MoonCraterHolder } from "components/Background/Moon/MoonCraters/styles.tsx";

interface MoonCraterProps {
  top: number;
  left: number;
  width: number;
  height: number;
  opacity: number;
}

function MoonCrater({ top, left, width, height, opacity }: MoonCraterProps) {
  const stylesArray = [
    `top: ${top}px`,
    `left: ${left}px`,
    `width: ${width}px`,
    `height: ${height}px`,
    `opacity: ${opacity}`,
  ];

  return <MoonCraterHolder $styles={stylesArray} />;
}

export default MoonCrater;
