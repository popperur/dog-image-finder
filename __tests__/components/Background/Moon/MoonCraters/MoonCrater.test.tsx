import "jest-styled-components";
import { render, RenderResult } from "@testing-library/react";
import { describe, it, expect, beforeEach } from "vitest";
import MoonCrater from "components/Background/Moon/MoonCraters/MoonCrater";

describe("MoonCrater component", () => {
  let rendered: RenderResult;

  beforeEach(() => {
    rendered = render(
      <MoonCrater top={10} left={20} width={40} height={60} opacity={0.5} />,
    );
  });

  it("matches the snapshot", () => {
    const { asFragment } = rendered;
    expect(asFragment()).toMatchSnapshot();
  });

  it("adds the styles to the styled component", () => {
    const { container } = rendered;

    const moonCraterHolder = container.firstChild;

    expect(moonCraterHolder).toHaveStyleRule("top", "10px");
    expect(moonCraterHolder).toHaveStyleRule("left", "20px");
    expect(moonCraterHolder).toHaveStyleRule("width", "40px");
    expect(moonCraterHolder).toHaveStyleRule("height", "60px");
    expect(moonCraterHolder).toHaveStyleRule("opacity", "0.5");
  });
});
