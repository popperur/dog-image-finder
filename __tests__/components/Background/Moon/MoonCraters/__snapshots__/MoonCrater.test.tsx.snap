// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MoonCrater component > matches the snapshot 1`] = `
<DocumentFragment>
  .c0 {
  position: absolute;
  background-color: #cadad6;
  border-radius: 50%;
  top: 10px;
  left: 20px;
  width: 40px;
  height: 60px;
  opacity: 0.5;
}

<div
    class="c0"
  />
</DocumentFragment>
`;
